force-app/main/default/classes/NPSP_Address.cls

```diff
@@ -87,7 +87,7 @@ public inherited sharing class NPSP_Address implements IAddress {
         )) {
             copyFromSObject(con, 'Mailing', con.npe01__Primary_Address_Type__c);
         } else {
-            copyFromSObject(con, 'Mailing', 'Mailing');
+            copyFromSObject(con, 'Mailing', null);
```

Comment:
Is the intent for `null` to not update the Address type? If I'm not mistaken though, this field was always populated with a default address type - or is that code being removed because it wasn't the original legacy behavior?

Reply:
These two changes together fix this bug: https://github.com/SalesforceFoundation/NPSP/issues/6522

Before, this field would be set to Home or Mailing even when there is no Address info on the Contact.  Now a blank Primary Address Type on the Contact is respected.  New installs will see blank respected, existing customers will see the default picklist value of Home since it can't be push upgraded.  I ran this by @jesslopez-sf who thought the changes are an improvement and move more toward expected behavior.

Reply:
Thank you for the explanation. Makes perfect sense. I forgot that the default value change won't be applied to existing customers.